pkg/cloud/aws/actuators/scope.go

```diff
@@ -34,6 +35,29 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/patch"
 )
 
+var (
+	sessionCache sync.Map
+)
+
+func sessionForConfig(config *aws.Config) (*session.Session, error) {
+	s, ok := sessionCache.Load(config.Region)
+	if ok {
+		return s.(*session.Session), nil
+	}
+
+	ns, err := session.NewSession(config)
+	if err != nil {
+		return nil, err
+	}
+
+	s, ok = sessionCache.LoadOrStore(config.Region, ns)
```

Comment:
Why not just `sessionCache.Store(config.Region, ns)` here? In the racy case we would be discarding a session anyway, so why not just discard the older session from the cache?

Reply:
Yeah maybe that's fine, I was trying to avoid having multiple sessions doing different calls to AWS. Truthfully without a lock on the function, that's not guaranteed, wdyt?

Reply:
If you want the guarantee, you need the mutex, yes.

Reply:
Fixed this as well